### PR TITLE
User can complete a shopping trip

### PR DESCRIPTION
**Pull request by kfrostare**
### User Story
```
As a resident 
In order to complete a trip
I would like to be able to interact with accepted requesters
```
## Changes proposed in this pull request:
* Creates button for trip completed
* Refactors show- logic for when trip is completed
* Creates total cost function for 
* Displays total cost to requester

## What I have learned working on this feature:
* How refactoring a larger component into smaller components can fix huge bug issues
* How to use ternaries in if- statements

## Screenshots (Client)
Screenshot 2020-04-19 at 21 38 42
Screenshot 2020-04-19 at 21 38 52
Screenshot 2020-04-19 at 21 39 33

## Lo-Fi's (Client)
image

### CODE REVIEW

File: screens/PongToPingDetails.jsx [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+    );
+    dispatch({
+      type: GET_TRIP_DETAILS,
+      payload: {
+        costSentMessage: response.data.message
+      }
+    });
+  }
+
+
```
Comment (emtalen) [inline screens/PongToPingDetails.jsx:53]: got an extra line here

File: screens/TripDetailsHeader.jsx [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+      >
+        <Text style={styles.buttonText} id="close-trip-button">
+          No More Pongs
+        </Text>
+      </TouchableHighlight>
+    )
+  }
+
+
+  return (
```
Comment (emtalen) [inline screens/TripDetailsHeader.jsx:47]: extra space

File: screens/PongToPingDetails.jsx [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+    backgroundColor: "#71B280",
+    justifyContent: "center",
+    alignItems: "center"
+  }
+});
+
+export { PongToPingDetails }
+
+
+
```
Comment (emtalen) [inline screens/PongToPingDetails.jsx]: lot of extra lines

File: screens/TripDetails.jsx [hunk 4]
```diff
@@ -10,45 +10,25 @@ import {
...
-    } else {
-      setCheck("checked");
-    }
-  };
+  const myPongs = useSelector(state => state.myPongs);
+  const pingId = useSelector(state => state.userTrip.id);
+  const completeTripMessage = useSelector(state => state.completeTripMessage)
 
   function Item({
-    pong_id,
```
Comment (emtalen) [inline screens/TripDetails.jsx:51]: redo this into an arrow function

**Review by emtalen — COMMENTED** [review]
Just a few comments but really good job you guys! 💯

File: screens/TripDetailsHeader.jsx [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+          closeTrip(pingId, userId, dispatch);
+        }}
+      >
+        <Text style={styles.buttonText} id="close-trip-button">
+          No More Pongs
+        </Text>
+      </TouchableHighlight>
+    )
+  }
+
```
Comment (pgaunitz) [inline screens/TripDetailsHeader.jsx:46]: extra line

File: screens/PongToPingDetails.jsx [hunk 6]
```diff
@@ -0,0 +1,248 @@
...
+      }
+    );
+    dispatch({
+      type: GET_TRIP_DETAILS,
+      payload: {
+        costSentMessage: response.data.message
+      }
+    });
+  }
+
```
Comment (pgaunitz) [inline screens/PongToPingDetails.jsx]: extra line

File: screens/PongToPingDetails.jsx [hunk 7]
```diff
@@ -0,0 +1,248 @@
...
+  let pong;
+  const pingId = useSelector(state => state.userTrip.id);
+  const costSentMessage = useSelector(state => state.costSentMessage);
+  const dispatch = useDispatch();
+  const [totalCost, setTotalCost] = useState()
+
+  let headers = JSON.parse(localStorage.getItem("J-tockAuth-Storage"));
+  const sendCostInformation = async (event, pongId) => {
+    event.preventDefault();
+    debugger
```
Comment (pgaunitz) [inline screens/PongToPingDetails.jsx]: debugger

**Comment by pgaunitz** [comment]
LG :rocket:

File: screens/PongToPingDetails.jsx (hunk 1 above)
Comment (kfrostare) [reply screens/PongToPingDetails.jsx:53]: done

File: screens/TripDetailsHeader.jsx (hunk 2 above)
Comment (kfrostare) [reply screens/TripDetailsHeader.jsx:47]: done

File: screens/PongToPingDetails.jsx (hunk 3 above)
Comment (kfrostare) [reply screens/PongToPingDetails.jsx]: done

File: screens/TripDetails.jsx (hunk 4 above)
Comment (kfrostare) [reply screens/TripDetails.jsx:51]: done, thank you for the help!

File: screens/TripDetailsHeader.jsx (hunk 5 above)
Comment (kfrostare) [reply screens/TripDetailsHeader.jsx:46]: done

File: screens/PongToPingDetails.jsx (hunk 6 above)
Comment (kfrostare) [reply screens/PongToPingDetails.jsx]: done

**Review by emtalen — APPROVED** [review]
Just think about to not change/format files that doesn't have anything to do with the actual PR, it is so many files changed then and it does not reflect the scope of the PR 😄

**Pull request merged**